plugin/lighthouse/lighthouse.go

```diff
@@ -25,17 +25,22 @@ var (
 var log = clog.NewWithPlugin("lighthouse")
 
 type Lighthouse struct {
-	Next           plugin.Handler
-	Fall           fall.F
-	Zones          []string
-	ttl            uint32
-	serviceImports *serviceimport.Map
-	endpointSlices *endpointslice.Map
-	clusterStatus  ClusterStatus
+	Next            plugin.Handler
+	Fall            fall.F
+	Zones           []string
+	ttl             uint32
+	serviceImports  *serviceimport.Map
+	endpointSlices  *endpointslice.Map
+	clusterStatus   ClusterStatus
+	endpointsStatus EndpointsStatus
 }
 
 type ClusterStatus interface {
 	IsConnected(clusterId string) bool
 }
 
+type EndpointsStatus interface {
+	IsHealthy(key, clusterId string) bool
```

Comment:
I suggest passing in name and namespace to keep key construction internal to the implementation.

```suggestion
	IsHealthy(name, namespace, clusterId string) bool
```